### PR TITLE
Submit BarebonesTitleCard & RetroTitleCard

### PR DESCRIPTION
**Pull request by YozoraXCII**

### CODE REVIEW

File: Yozora/BarebonesTitleCard.py [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+
+from num2words import num2words
+
+from modules.CardType import CardType
+from modules.Debug import log
+
+class BarebonesTitleCard(CardType):
+    """
+    This class describes a type of ImageMaker that produces title cards in the
+    theme of Star Wars cards as designed by reddit user /u/Olivier_286.
```
Comment (CollinHeist) [inline Yozora/BarebonesTitleCard.py:13]:
```suggestion
```

File: Yozora/BarebonesTitleCard.py [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+    """
+
+    """Directory where all reference files used by this card are stored"""
+    REF_DIRECTORY = Path(__file__).parent / 'ref' / 'barebones'
```
Comment (CollinHeist) [inline Yozora/BarebonesTitleCard.py]:
```suggestion
    REF_DIRECTORY = Path(__file__).parent.parent / 'ref' / 'barebones'
```

File: Yozora/BarebonesTitleCard.py [hunk 3]
```diff
@@ -0,0 +1,303 @@
+from pathlib import Path
+from re import match
+
+from num2words import num2words
+
+from modules.CardType import CardType
+from modules.Debug import log
```
Comment (CollinHeist) [inline Yozora/BarebonesTitleCard.py:7]:
```suggestion
from modules.Debug import log
from modules.RemoveFile import RemoteFile
```

File: Yozora/BarebonesTitleCard.py [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+
+    """Characteristics for title splitting by this class"""
+    TITLE_CHARACTERISTICS = {
+        'max_line_width': 16,   # Character count to begin splitting titles
+        'max_line_count': 5,    # Maximum number of lines a title can take up
+        'top_heavy': True,      # This class uses top heavy titling
+    }
+
+    """Characteristics of the default title font"""
+    TITLE_FONT = str((REF_DIRECTORY/'Montserrat-Bold.ttf').resolve())
```
Comment (CollinHeist) [inline Yozora/BarebonesTitleCard.py]:
```suggestion
    TITLE_FONT = str(RemoteFile('Yozora', 'ref/barebones/Montserrat-Bold.ttf'))
```

File: Yozora/BarebonesTitleCard.py [hunk 5]
```diff
@@ -0,0 +1,303 @@
...
+    TITLE_COLOR = '#FFFFFF'
+    FONT_REPLACEMENTS = {}
+
+    """Characteristics of the episode text"""
+    EPISODE_TEXT_FORMAT = 'EPISODE {episode_number}'
+    EPISODE_TEXT_COLOR = '#FFFFFF'
+    EPISODE_TEXT_FONT = REF_DIRECTORY / 'Montserrat-SemiBold.ttf'
```
Comment (CollinHeist) [inline Yozora/BarebonesTitleCard.py]:
```suggestion
    EPISODE_TEXT_FONT = RemoteFile('Yozora', 'ref/barebones/Montserrat-SemiBold.ttf'
```

File: Yozora/BarebonesTitleCard.py [hunk 6]
```diff
@@ -0,0 +1,303 @@
...
+            self.episode_prefix = 'PART'
+            modified_text = modified_text.replace('PART', '')
+
+        try:
+            # Only digit episode text remains, return as a number (i.e. "two")
+            return num2words(int(modified_text.strip())).upper()
+        except ValueError:
+            # Not just a digit, return as-is
+            return modified_text.strip()
+
```
Comment (CollinHeist) [inline Yozora/BarebonesTitleCard.py]:
```suggestion
```

File: Yozora/RetroTitleCard.py [hunk 7]
```diff
@@ -0,0 +1,333 @@
...
+from re import findall
+
+from modules.CardType import CardType
+from modules.Debug import log
+from modules.RemoteFile import RemoteFile
+
+class RetroTitleCard(CardType):
+    """
+    This class describes lyonza's CardType based on Wvdh's
+    "WhiteTextBroadcast" card to show SxxExx format instead of absolute numbering
```
Comment (CollinHeist) [inline Yozora/RetroTitleCard.py]:
```suggestion
```

File: Yozora/RetroTitleCard.py [hunk 8]
```diff
@@ -0,0 +1,333 @@
...
+    """
+
+
+    """Directory where all reference files used by this card are stored"""
+    REF_DIRECTORY = Path(__file__).parent / 'ref' / 'retro'
```
Comment (CollinHeist) [inline Yozora/RetroTitleCard.py]:
```suggestion
    REF_DIRECTORY = Path(__file__).parent.parent / 'ref' / 'retro'
```

File: Yozora/RetroTitleCard.py [hunk 9]
```diff
@@ -0,0 +1,333 @@
...
+    USES_SEASON_TITLE = True
+
+    """Standard class has standard archive name"""
+    ARCHIVE_NAME = 'Retro Style'
+    
+    EPISODE_TEXT_FORMAT = "S{season_number:02}E{episode_number:02}"
+    
+    """Source path for the gradient image overlayed over all title cards"""
+    __GRADIENT_IMAGE_PLAY = REF_DIRECTORY / 'gradient_play.png'
+    __GRADIENT_IMAGE_REWIND = REF_DIRECTORY / 'gradient_rewind.png'
```
Comment (CollinHeist) [inline Yozora/RetroTitleCard.py]:
```suggestion
    __GRADIENT_IMAGE_PLAY = RemoteFile('Yozora', 'ref/retro/gradient_play.png')
    __GRADIENT_IMAGE_REWIND = RemoteFile('Yozora', 'ref/retro/gradient_rewind.png')
```

File: Yozora/RetroTitleCard.py [hunk 10]
```diff
@@ -0,0 +1,333 @@
...
+
+    """Default fonts and color for series count text"""
+    SEASON_COUNT_FONT = REF_DIRECTORY / 'retro.ttf'
+    EPISODE_COUNT_FONT = REF_DIRECTORY / 'retro.ttf'
```
Comment (CollinHeist) [inline Yozora/RetroTitleCard.py]:
```suggestion
    SEASON_COUNT_FONT = RemoteFile('Yozora', 'ref/retro/retro.ttf')
    EPISODE_COUNT_FONT = RemoteFile('Yozora', 'ref/retro/retro.ttf')
```

File: Yozora/RetroTitleCard.py [hunk 11]
```diff
@@ -0,0 +1,333 @@
...
+
+    """Paths to intermediate files that are deleted after the card is created"""
+    __SOURCE_WITH_GRADIENT = CardType.TEMP_DIR / 'source_gradient.png'
+    __GRADIENT_WITH_TITLE = CardType.TEMP_DIR / 'gradient_title.png'
+    __SERIES_COUNT_TEXT = CardType.TEMP_DIR / 'series_count_text.png'
+
+    __slots__ = ('source_file', 'output_file', 'title',
+                 'episode_text', 'font', 'font_size', 'title_color',
+                 'hide_season', 'blur', 'vertical_shift', 'interline_spacing',
+                 'kerning', 'stroke_width')
```
Comment (CollinHeist) [inline Yozora/RetroTitleCard.py]:
```suggestion
                 'watched', 'kerning', 'stroke_width')
```

**Review by CollinHeist — CHANGES_REQUESTED** [review]
Looks really good! A few things:

- [ ] Looks like you're not using Motserrat-Regular.ttf, so we should delete that
- [ ] Add the new cards to the main README

**Comment by YozoraXCII** [comment]
> Looks really good! A few things:
> 
> * [ ]  Looks like you're not using Motserrat-Regular.ttf, so we should delete that
> * [ ]  Add the new cards to the main README

I believe I have addressed both of these

**Review by CollinHeist — APPROVED** [review]
Looks good - nicely done! Love the cards

**Pull request merged**